docker-compose/docker-compose.yaml

```diff
@@ -2,23 +2,57 @@ version: '2.4'
 services:
   # Description: Acts as a reverse proxy for all of the sourcegraph-frontend instances
   #
-  # Disk: none
+  # Disk: 1GB / persistent SSD
   # Ports exposed to other Sourcegraph services: none
-  # Ports exposed to the public internet: 7080 (HTTP) and/or 7443 (HTTPS)
-  #
-  # Note: nginx/nginx/sourcegraph_backend.conf lists all of the sourcegraph-frontend
-  # network addresses that Nginx should proxy. nginx/nginx/sourcegraph_backend.conf
-  # needs to be updated with new addresses when scaling the number of soucegraph-frontend
-  # replicas.
-  nginx:
-    container_name: nginx
-    image: 'index.docker.io/library/nginx:1.17.7@sha256:8aa7f6a9585d908a63e5e418dc5d14ae7467d2e36e1ab4f0d8f9d059a3d071ce'
+  # Ports exposed to the public internet: 80 (HTTP) and 443 (HTTPS)
+  #
+  # Sourcegraph ships with a few builtin templates that cover some common HTTP/HTTPS configurations:
+  # - HTTP only (default)
+  # - HTTPS with Let's Encrypt (staging/production - strongly recommended that you test with the staging
+  #                            configuration first, see caddy/builtins/https.lets-encrypt-staging.Caddyfile
+  #                            for more info)
+  # - HTTPS with custom certificates
+  #
+  # Follow the directions in the comments below to swap between these configurations.
+  #
+  # If none of these built-in configurations suit your needs, then you can create your own Caddyfile by referencing
+  # the official documentation: https://caddyserver.com/docs/caddyfile
+  caddy:
+    container_name: caddy
+    image: 'index.docker.io/caddy/caddy:alpine@sha256:b0f0a763527ca11a555bc5b5f942fa2d84ae0c90b0b60e47d0426ff884dd10ad'
     cpus: 1
     mem_limit: '1g'
```

Comment:
I suspect we actually should give Caddy 4 CPU and 4GB memory as limits instead. It's only going to use as much as it needs anyway (i.e. not like search services which do not directly correlate requests with resources) and if we don't this could act as a choke point for the entire app.